### ISSUE TITLE
QScrollArea height error.

### ISSUE THREAD

**Issue by LeonLoong** [ColinDuquesnoy/QDarkStyleSheet#216]
### Environment
- QDarkStyle: 2.8
- OS: Windows
- PySide: 1.2.4
- QtPy: 1.9.0

### Language
- Python2.7

[Description of the issue]
When I set  QDarkStyleSheet for my UI, the QToolButton is become mini after uncollapsed.
Below is GIF reference for my issues. Thanks!

### Actual Result
![with_qds2](https://user-images.githubusercontent.com/11975617/74218613-760ce600-4ce5-11ea-91f7-af3ed4be2769.gif)
The tool button is became mini after dis-collapsed.

### Expected Results
![without_qds](https://user-images.githubusercontent.com/11975617/74218394-fa129e00-4ce4-11ea-9324-51870d4a5512.gif)
This is result without QDarkStyle.

# code to reproduce the problem

```
from PySide import QtGui, QtCore
import qdarkstyle

class Spoiler(QtGui.QWidget):

    def __init__(self, parent=None, title='', animationDuration=300):

        super(Spoiler, self).__init__(parent=parent)
        self.groupBox = QtGui.QGroupBox(self)
        self.groupBox.setGeometry(QtCore.QRect(210, 30, 95, 134))
        self.groupBox.setObjectName("groupBox")
        self.verticalLayout = QtGui.QVBoxLayout(self.groupBox)
        self.verticalLayout.setObjectName("verticalLayout")
        self.pushButton3 = QtGui.QPushButton(self.groupBox)
        icon = QtGui.QIcon()

        self.pushButton3.setIcon(icon)
        self.pushButton3.setObjectName("pushButton3")
        self.verticalLayout.addWidget(self.pushButton3)
        self.pushButton2 = QtGui.QPushButton(self.groupBox)
        self.pushButton2.setObjectName("pushButton2")
        self.verticalLayout.addWidget(self.pushButton2)
        self.label_2 = QtGui.QLabel(self.groupBox)

        self.label_2.setObjectName("label_2")
        self.verticalLayout.addWidget(self.label_2)
        self.pushButton = QtGui.QPushButton(self.groupBox)
        self.pushButton.setIcon(icon)
        self.pushButton.setObjectName("pushButton")
        self.verticalLayout.addWidget(self.pushButton)

        self.animationDuration = 300
        self.toggleAnimation = QtCore.QParallelAnimationGroup()
        self.contentArea = QtGui.QScrollArea()
        self.headerLine = QtGui.QFrame()
        self.toggleButton = QtGui.QToolButton()
        self.mainLayout = QtGui.QGridLayout()

        toggleButton = self.toggleButton
        toggleButton.setStyleSheet("QToolButton { border: none; }")
        toggleButton.setToolButtonStyle(QtCore.Qt.ToolButtonTextBesideIcon)
        toggleButton.setArrowType(QtCore.Qt.RightArrow)
        toggleButton.setText(str(title))
        toggleButton.setCheckable(True)
        toggleButton.setChecked(False)

        headerLine = self.headerLine
        headerLine.setFrameShape(QtGui.QFrame.HLine)
        headerLine.setFrameShadow(QtGui.QFrame.Sunken)
        headerLine.setSizePolicy(QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Maximum)

        self.contentArea.setStyleSheet(
            "QScrollArea { background-color: white; border: none; }")
        self.contentArea.setSizePolicy(
            QtGui.QSizePolicy.Expanding, QtGui.QSizePolicy.Fixed)
        # start out collapsed
        self.contentArea.setMaximumHeight(0)
        self.contentArea.setMinimumHeight(0)
        # let the entire widget grow and shrink with its content
        toggleAnimation = self.toggleAnimation
        toggleAnimation.addAnimation(
            QtCore.QPropertyAnimation(self, b"minimumHeight"))
        toggleAnimation.addAnimation(
            QtCore.QPropertyAnimation(self, b"maximumHeight"))
        toggleAnimation.addAnimation(
            QtCore.QPropertyAnimation(self.contentArea, b"maximumHeight"))
        # don't waste space
        mainLayout = self.mainLayout
        mainLayout.setVerticalSpacing(0)
        mainLayout.setContentsMargins(0, 0, 0, 0)
        row = 0
        mainLayout.addWidget(self.toggleButton, row, 0,
                             1, 1, QtCore.Qt.AlignLeft)
        mainLayout.addWidget(self.headerLine, row, 2, 1, 1)
        row += 1
        mainLayout.addWidget(self.contentArea, row, 0, 1, 3)
        self.setLayout(self.mainLayout)

        def start_animation():
            checked = self.toggleButton.isChecked()
            arrow_type = QtCore.Qt.DownArrow if checked else QtCore.Qt.RightArrow
            direction = QtCore.QAbstractAnimation.Forward if checked else QtCore.QAbstractAnimation.Backward
            toggleButton.setArrowType(arrow_type)
            self.toggleAnimation.setDirection(direction)
            self.toggleAnimation.start()

        self.toggleButton.clicked.connect(start_animation)

    def setContentLayout(self, contentLayout):
        # Not sure if this is equivalent to self.contentArea.destroy()
        self.contentArea.destroy()
        self.contentArea.setLayout(contentLayout)
        collapsedHeight = self.sizeHint().height() - self.contentArea.maximumHeight()
        contentHeight = contentLayout.sizeHint().height()
        for i in range(self.toggleAnimation.animationCount()-1):
            spoilerAnimation = self.toggleAnimation.animationAt(i)
            spoilerAnimation.setDuration(self.animationDuration)
            spoilerAnimation.setStartValue(collapsedHeight)
            spoilerAnimation.setEndValue(collapsedHeight + contentHeight)
        contentAnimation = self.toggleAnimation.animationAt(
            self.toggleAnimation.animationCount() - 1)
        contentAnimation.setDuration(self.animationDuration)
        contentAnimation.setStartValue(0)
        contentAnimation.setEndValue(contentHeight)


if __name__ == "__main__":
    import sys

    app = QtGui.QApplication(sys.argv)

    ui = Spoiler()
    ui.setContentLayout(ui.verticalLayout)
    ui.setStyleSheet(qdarkstyle.load_stylesheet_pyside())
    ui.show()

    sys.exit(app.exec_())

```

**Comment by dpizetta** [comment]
Tks for reporting :)

**Comment by dpizetta** [comment]
Hi @LeonLoong, you need to work with pyside v1.2? Could you try with pyside 2? Tks. Changin g to milestone 2.9 as I couldn't solve it for now.

**Comment by dpizetta** [comment]
I've been testing for some time, and it seems to be some combination of widgets and structure that causes this in dark mode. It may take a while to fix as I need to found what combination causes it.

**Comment by LeonLoong** [comment]
> Hi @LeonLoong, you need to work with PySide v1.2? Could you try with PySide 2? Tks. Changin g to milestone 2.9 as I couldn't solve it for now.

Hi @dpizetta, actually I using QDarkStyle for creating a small tool for a Maya 3D software, and it's built-in PySide v1.2 by default. It's ok I can test QDarkStyle with PySide2 when I am free.

**Comment by LeonLoong** [comment]
> I've been testing for some time, and it seems to be some combination of widgets and structure that causes this in dark mode. It may take a while to fix as I need to found what combination causes it.

It's ok, take your time.

**Comment by LeonLoong** [comment]
> I've been testing for some time, and it seems to be some combination of widgets and structure that causes this in dark mode. It may take a while to fix as I need to found what combination causes it.

It's ok, take your time.

**Comment by LeonLoong** [comment]
> I've been testing for some time, and it seems to be some combination of widgets and structure that causes this in dark mode. It may take a while to fix as I need to found what combination causes it.

It's ok, take your time.

**Comment by dpizetta** [comment]
@LeonLoong any news about it? We will not be able to provide a fix if only breaks in python 2.7 (there is no maintenance anymore) and PySide1. If needed, please reopen this in the future, and thanks for reporting.
